OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -37,14 +38,18 @@ public ReturnToBase(Actor self, bool abortOnResupply, Actor dest = null, bool al
 			aircraft = self.Trait<Aircraft>();
 			aircraftInfo = self.Info.TraitInfo<AircraftInfo>();
 			repairableInfo = self.Info.TraitInfoOrDefault<RepairableInfo>();
+			rearmable = self.TraitOrDefault<Rearmable>();
 		}
 
 		public static Actor ChooseResupplier(Actor self, bool unreservedOnly)
 		{
-			var rearmBuildings = self.Info.TraitInfo<AircraftInfo>().RearmBuildings;
+			var rearmInfo = self.Info.TraitInfoOrDefault<RearmableInfo>();
+			if (rearmInfo == null)
```

Comment:
Can we check if `rearmable` is `null` here and use `rearmabl.Info` later on instead of another trait query?